python/dgl/distributed/graph_services.py

```diff
@@ -404,7 +394,33 @@ def issue_remote_req(node_ids):
     def local_access(local_g, partition_book, local_nids):
         return _sample_neighbors(local_g, partition_book, local_nids,
                                  fanout, edge_dir, prob, replace)
-    return _distributed_access(g, nodes, issue_remote_req, local_access)
+    frontier = _distributed_access(g, nodes, issue_remote_req, local_access)
+    if len(gpb.etypes) > 1:
+        etype_ids, frontier.edata[EID] = gpb.map_to_per_etype(frontier.edata[EID])
+        src, dst = frontier.edges()
+        etype_ids, idx = F.sort_1d(etype_ids)
```

Comment:
Will sort speed things up?

Reply:
currently, RGCN also requires to sort edges based on edge types.
we can avoid the sorting if `sample_neighbors` returns heterogeneous graphs directly.